sdk/eventhub/azure-eventhub/tests/livetest/asynctests/test_auth_async.py

```diff
@@ -33,7 +33,7 @@ async def test_client_secret_credential_async(live_eventhub):
         batch.add(EventData(body='A single message'))
         await producer_client.send_batch(batch)
 
-    def on_event(partition_context, event):
+    async def on_event(partition_context, event):
```

Comment:
Was this test always broken?

Reply:
yea.. the funny thing is that the test passed because the sync callback is executed.
in the while, await a sync call logs a error/warning if on_error is not callback which is the expected behavior..

I will make a PR to update the test case